torch/optim/optimizer.py

```diff
@@ -72,6 +74,16 @@ def __repr__(self):
         format_string += ')'
         return format_string
 
+    def _step_wrapper(self):
+        """Wrap self.step under profiler. In order to avoid of instrumenting each sub-class."""
+        func = self.step
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            profile_label = str.format("{}#{}", func.__qualname__, "Optimizer.step")
```

Comment:
I prefer `TYPE#CLASS.METHOD`. It is a trade-off solution between us...
Because we don't define "TYPE" as CLASS.METHOD, it can be any string we like. For example, "enumerate(DataLoader)" may be a good string as "TYPE" for user to see. Then we can't discard the METHOD name, so I append it at tail.